### ISSUE TITLE
Incorrect warning issued when call/apply used outside parens

### ISSUE THREAD

**Issue by goatslacker** [douglascrockford/JSLint#101]
```
(function (that) {
    // that
}).call(this);
```

Will give me a `bad_wrap` warning when it should give me a `move_invocation` warning instead.

Patch can be found here:

https://gist.github.com/1819655
